src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -1700,4 +1700,36 @@ class Global(var currentSettings: Settings, var reporter: Reporter)
 
 object Global {
   def apply(settings: Settings, reporter: Reporter): Global = new Global(settings, reporter)
+
+  def apply(settings: Settings): Global = new Global(settings, reporter(settings))
+
+  private def reporter(settings: Settings): Reporter = {
+    def fail(msg: String) = {
+      settings.errorFn(msg)
+      //new ConsoleReporter(settings)  // fall back?
+      throw FatalError(msg)
+    }
+    try {
+      val loader = /*ScalaClassLoader*/ {
+        val ccl = Thread.currentThread.getContextClassLoader
+        if (ccl != null) ccl else getClass.getClassLoader
+      }
+      //val clazz = loader tryToInitializeClass settings.reporter.value getOrElse ???
+      val clazz = loader loadClass settings.reporter.value
+      if (classOf[Reporter] isAssignableFrom clazz) {
+        val ctor  = clazz.getConstructor(classOf[Settings])    // scala.tools.nsc.Settings
+        (ctor newInstance settings).asInstanceOf[Reporter]
+      } else {
+        fail(s"Not a ${classOf[Reporter].getName}: ${settings.reporter.value}")
+      }
+    } catch {
+      case _: ClassNotFoundException =>
+        fail(s"Class not found: ${settings.reporter.value}")
+      case e: NoSuchMethodException =>
+        fail(s"Constructor must take an instance of scala.tools.nsc.Settings: ${settings.reporter.value}")
+      case e @ (_: IllegalAccessException | _: InstantiationException |
```

Comment:
Thx, I may also try to encapsulate, as suggested by the commented use of ScalaClassLoader.